README.md

```diff
@@ -21,88 +20,247 @@ and how you can use they in your project.
+
+When creating a `ScreenRoute` it can receive a optional generic type that will be considered as the type of the value
+to be returned by this Route when popped.
 
 Screen Options:
 
 - builder
 - screenType
-- deepLink
-- wrapperFn
+- wrapper
+
+FlowRoute is just a subclass of the `ScreenRoute` that may receive a `Nuvigator` instead of a builder function. Using
+the FlowRoute will allow Nuvigator to properly create all nested Nuvigator routing methods correctly.
+
+When using `FlowRoute` you **SHOULD** declare the generic type of the `Router` that is being used by it's Nuvigator. This
+is required so the static analysis is able to properly generate the code when running the builder.
+
+## Creating Routers
 
-### BaseRouter
 
-Basic Usage:
+Defining a new Router will probably be the what you will do the most when working with Nuvigator, so understanding how
+to do it properly is important. A Router class is a class that should extend a `BaseRouter` and annotate the class with 
+the `@NuRouter` annotation.
 
+### Defining Routes
+ 
+Inside your `MyCustomRouter` you can define your routes, each route will be represented by a
+method annotated with the `@NuRoute` annotation. Those methods should return a `ScreenRoute` or a `FlowRoute`, and can
+receive any number of **named** arguments, those will be used as the Arguments that should/can be passed to your route
+when navigating to it.
+
+Example:
 ```dart
-class ChatRouter extends BaseRouter {
+@NuRouter()
+class MyCustomRouter extends BaseRouter {
+  
+  @NuRoute()
+  ScreenRoute firstScreen({String argumentHere}) => ScreenRoute(
+    builder: (context) => MyFirstScreen(context),
+  ); 
+  
   @override
-  Map<String, ScreenRoute> get screensMap => {
-        'home': ScreenRoute(
-            builder: (context) => ChatHomeScreen(context),
-            screenType: materialScreenType,
-            deepLink: '/myGroup/test/', 
-          ),
-      };
+  Map<RouteDef, ScreenRouteBuilder> get screensMap  => _$myCustomScreensMap(this); // Will be created by generated code
+
 }
 ```
 
-Options:
+### DeepLinks
 
-- `screensMap`
+The `NuRoute` annotation may receive some options, in special, the `deepLink` option is used to declare a deepLink that
+will be used to access this route. The usage is the same when using a `FlowRoute`, the only changes will be in the generated
+code, that will take in account the nested nuvigator.
 
-A Map<String, Screen> that contains the mapping between RouteNames and Screens.
+DeepLinks support both pathParameters (using URL templates) and queryParameters. Those parameters will be passed to your
+Route as arguments as Strings, so keep that in mind when declaring the Route arguments.
 
-- `deepLinkPrefix`
+Example:
+```dart
+@NuRoute(deepLink: 'myRoute/:argumentName')
+```
 
-A String, that will be used as prefix for every entry key at the `deepLinksMaps`.
+_Obs_: DeepLinks present in routers used in nested Nuvigators inside `FlowRoute` will **NOT** be considered and are no 
```

Comment:
DeepLinks presented in routers that are used in nested Nuvigators (inside a `FlowRoute), will **NOT** be considered and are no ...